### DISCUSSION TITLE
Loading SMT2 from files in Python binding

### DISCUSSION THREAD

**Discussion by exokortex** [bitwuzla/bitwuzla#46] [General] [answered]
Bitwuzla can parse files via `bitwuzla_parse` and `bitwuzla_parse_format`, but it appears that those functions are not used in the Python binding (or rather [commented out](https://github.com/bitwuzla/bitwuzla/blob/1230d80a5275ddf525da9c218c2bcd29c3130e49/src/api/python/bitwuzla_api.pxd#L347)).

Are there plans to bring this functionality to the python bindings?

**Comment by mpreiner** [comment]
We are currently working on a complete rewrite of Bitwuzla, this will be supported in the first public release coming in the next couple of months.

**Comment by aniemetz** [comment, marked as answer]
@exokortex Bitwuzla now supports parsing via its Python bindings (e.g., https://github.com/bitwuzla/bitwuzla/blob/main/examples/python/parse.py).
